lib/ansible/modules/monitoring/zabbix/zabbix_template.py

```diff
@@ -244,32 +447,49 @@ def main():
                          result="Successfully delete template %s" %
                          template_name)
 
-    child_template_ids = []
-    if link_templates:
-        child_template_ids = template.get_template_ids(link_templates)
-
-    clear_template_ids = []
-    if clear_templates:
-        clear_template_ids = template.get_template_ids(clear_templates)
-
-    group_ids = template.get_group_ids_by_group_names(template_groups)
-    if not group_ids:
-        module.fail_json(msg='Template groups not found: %s' %
-                         str(template_groups))
+    elif state == "dump":
+        if not template_ids:
+            module.fail_json(msg='Template not found: %s' % template_name)
+        module.exit_json(changed=True, msg=existing_template_json)
```

Comment:
Why is changed=True here? Nothing should have changed on Zabbix's side...

Reply:
It may make sense to put the dump into a variable of its own instead of msg (which ordinarily contains user-readable messages). Maybe template_json?

Reply:
Done. Also I changed the return variable to `template_json` instead of `msg` so the dump is named more sensibly in the return.

Reply:
data is too generic, and template_json was the first thing that came to mind. I'm not married to it though.